connectors/sources/salesforce.py

```diff
@@ -42,31 +48,112 @@ def session(self):
         )
         resp_json = await resp.json()
         self.token = resp_json["access_token"]
         self.token_issued_at = resp_json["issued_at"]
+        self.token_refresh_enabled = enable_refresh
+        self._logger.debug("Salesforce token retrieved.")
 
-    async def ping(self):
-        await self._api_request(self.base_url, "head")
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+    )
+    async def get_accounts(self):
+        # TODO prepare cache
+
+        query_builder = SalesforceSoqlBuilder("Account")
+        query_builder.with_id()
+        query_builder.with_default_metafields()
+        query_builder.with_fields(
+            self._select_queryable_fields(
+                [
+                    "Name",
+                    "Description",
+                    "BillingAddress",
+                    "Type",
+                    "Website",
+                    "Rating",
+                    "Department",
+                ]
+            )
+        )
+        query = query_builder.build()
 
-    async def close_session(self):
-        if self.session is not None:
-            await self.session.close()
-            del self.session
+        # TODO handle pagination
+        resp = await self._yield_non_bulk_query_pages(query)
+        resp_json = await resp.json()
+        for record in resp_json.get("records", []):
+            record["_id"] = record["Id"]
+            yield record
+
+    async def _yield_non_bulk_query_pages(self, soql_query):
+        return await self._api_request(
+            f"{self.base_url}{QUERY_ENDPOINT}",
+            "get",
+            headers=self._auth_headers(),
+            params={"q": soql_query},
+        )
+
+    def _select_queryable_fields(self, fields):
+        # TODO check if fields queryable first from cache
+        return fields
+
+    def _token_cleanup(self):
+        self.token = None
+        self.token_issued_at = None
+        self.token_refresh_enabled = False
 
     def _auth_headers(self):
         return {"authorization": f"Bearer {self.token}"}
 
-    async def _api_request(self, url, method, data=None):
-        return await getattr(self.session, method)(url=url, data=data)
+    async def _api_request(self, url, method, headers=None, data=None, params=None):
+        # TODO improve error handling
+        return await getattr(self.session, method)(
+            url=url, headers=headers, data=data, params=params
+        )
+
+
+class SalesforceSoqlBuilder:
+    def __init__(self, table):
+        self.table_name = table
+        self.fields = []
+
+    def with_id(self):
+        self.fields.append("Id")
+
+    def with_default_metafields(self):
+        self.fields.extend(["CreatedDate", "LastModifiedDate"])
+
+    def with_fields(self, fields):
+        self.fields.extend(fields)
+
+    def build(self):
+        select_columns = ",\n".join(set(self.fields))
+
+        query_lines = [f"SELECT {select_columns}", f"FROM {self.table_name}"]
```

Comment:
I personally liked how it was before - having a line for each construct as you do in SQL feels naturally nicer. It's not a suggestion to change, but a comment on @praveen-elastic's suggestion - some code is IMO better when you don't try to unite it into one-liners or similar constructs, e.g. compare:

```
query_lines = []
query_lines << f"SELECT {columns}"
query_lines << f"FROM {table}"
if where_statement:
  query_lines << f"WHERE {where_statement}"
query_lines << f"ORDER BY {order}"
```

vs

```
query_lines = [f"SELECT {columns}", f"FROM {table}"]
if where_statement:
  query_lines << f"WHERE {where_statement}"
query_lines << f"ORDER BY {order}"
```

It can be changed later when new order bys and other things are added later, if you feel it's worth changing :)